notebook/notebookapp.py

```diff
@@ -2294,9 +2317,10 @@ def list_running_servers(runtime_dir=None):
             with io.open(os.path.join(runtime_dir, file_name), encoding='utf-8') as f:
                 info = json.load(f)
 
-            # Simple check whether that process is really still running
+            # active check whether that process is really available via real HTTP request
             # Also remove leftover files from IPython 2.x without a pid field
-            if ('pid' in info) and check_pid(info['pid']):
+            response = kernel_request(info, path='/login')
+            if response.body.find(b'<title>Jupyter Notebook</title>') > 0:
```

Comment:
> When running on my system, I find a title of: `<title>Home Page - Select or create a notebook</title>`. Not sure if there can be others, but both of these should be tolerated.

i have assigned two kind of values:
'<title>Home Page - Select or create a notebook</title' for UNIX socket: "jupyter-notebook --sock=... ";
'<title>Jupyter Notebook</title>' for TCP socket;

> 1. Can these titles be localized?  If so, that will be problematic.  Perhaps there's another indicator that can be used?

till now, all tests passed;

> 2. When invoked when no server is running, it can take a while to get the exception back (or perhaps `None` based on previous comment).  If there are multiple files to process, this can take a while.

one step one footprint;
i can make multiple commits;
you think is useful. we can works together on more commits;

> 3. Since there can be multiple files referencing the same port, but different PIDs, results can be a little misleading.  For example, should the subsequent `/api/shutdown` request not shut the server down (for whatever reason), the fallback logic to send `SIGTERM` to the pid, could result in the termination of the wrong application, and we're back to where we started.

this commit only one footprint;
you consideration is really useful.

we can define server_info_file naming rule; for example: hash(str(self.sock) + str(self.port));
then reuse port will overwrite previous connection file in "self.write_server_info_file()":
further commits on persistent file naming rule; mark here. 
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    info_file = Unicode()                                                                                                                                                   
                                                                                                                                                                            
    @default('info_file')                                                                                                                                                   
    def _default_info_file(self):                                                                                                                                           
        info_file = "nbserver-%s.json" % os.getpid()                                                                                                                        
        return os.path.join(self.runtime_dir, info_file)                                                                                                                    
                                                                                                                                                                            
    browser_open_file = Unicode()                                                                                                                                           
                                                                                                                                                                            
    @default('browser_open_file')                                                                                                                                           
    def _default_browser_open_file(self):                                                                                                                                   
        basename = "nbserver-%s-open.html" % os.getpid()                                                                                                                    
        return os.path.join(self.runtime_dir, basename)

Reply:
[master a4aae8fe6] consistent persistent file naming rule and same version check rule
+            response = kernel_request(info, path='/api')
+            if response.body == api_version_json_bytes:
                 yield info